File: Python/programmers/etc/level3/인사고과.py
def solution(scores):
    answer = 1
    
    target = scores[0]
    scores.sort(key=lambda t: (-t[0], t[1]))
    
    _min = 0
    for score in scores:
        if target[0] < score[0] and target[1] < score[1]:
            return -1
        if _min <= score[1]:
            if sum(target) < sum(score):
                answer += 1
            _min = score[1]
            
    return answer


scores = [[2,2],[1,4],[3,2],[3,2],[2,1]]
print(solution(scores)) # 4

File: Java/programmers/etc/level3/인사고과.java
package CodeTest.Java.programmers.etc.level3;

import java.util.*;

public class 인사고과 {
    public int solution(int[][] scores) {
        int answer = 1;
        int[] target = scores[0];

        Arrays.sort(scores, (o1, o2) -> {
            return (o1[0]==o2[0]) ? o1[1]-o2[1] : o2[0]-o1[0]; 
        });

        int _min = 0;
        for (int[] score : scores) {
            if (target[0] < score[0] && target[1] < score[1]) return -1;
            if (_min <= score[1]) {
                if (target[0]+target[1] < score[0]+score[1]) answer++;
                _min = score[1];
            }
        }

        return answer;
    }
}
